src/overlays/actors/ovl_En_Xc/z_en_xc.c

```diff
@@ -77,479 +48,2375 @@ static ColliderCylinderInitType1 sCylinderInit = {
+
+    if (this->attached) {
+        attachedPos = &this->attached->world.pos;
+        if (!this) {}
+        attachedPos->x = npcAction->startPos.x;
+        attachedPos->y = npcAction->startPos.y;
+        attachedPos->z = npcAction->startPos.z;
+    }
+}
+
+void EnXc_DestroyFlame(EnXc* this) {
+    if (this->attached) {
+        Actor_Kill(this->attached);
+        this->attached = NULL;
+    }
+    Actor_Kill(&this->actor);
+}
+
+void EnXc_InitFlame(EnXc* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s16 sceneNum = globalCtx->sceneNum;
+    static s32 D_80B41DA8 = 1;
+
+    if (sceneNum == SCENE_SPOT17) {
+        CsCmdActorAction* npcAction = EnXc_GetCsCmd(globalCtx, 0);
+        if (npcAction) {
+            s32 action = npcAction->action;
+            if (D_80B41DA8 != action) {
+                if (action != 1) {
+                    EnXc_SpawnFlame(this, globalCtx);
+                }
+
+                if (action == 1) {
+                    EnXc_DestroyFlame(this);
+                }
+
+                D_80B41DA8 = action;
+            }
+
+            EnXc_SetupFlamePos(this, globalCtx);
+        }
+    }
+}
+
+void func_80B3D48C(EnXc* this, GlobalContext* globalCtx) {
+    CutsceneContext* csCtx = &globalCtx->csCtx;
+    CsCmdActorAction* linkAction = csCtx->linkAction;
+    s16 yaw;
+
+    if (linkAction) {
+        yaw = (u16)linkAction->rot.y + 0x8000;
+    } else {
+        Player* player = PLAYER;
+        yaw = player->actor.world.rot.y + 0x8000;
+    }
+
+    this->actor.world.rot.y = yaw;
+    this->actor.shape.rot.y = this->actor.world.rot.y;
+}
+
+AnimationHeader* EnXc_GetCurrentHarpAnim(GlobalContext* globalCtx, s32 index) {
+    AnimationHeader* animation = &gShiekPlayingHarp5Anim;
+    CsCmdActorAction* npcAction = EnXc_GetCsCmd(globalCtx, index);
+
+    if (npcAction) {
+        u16 action = npcAction->action;
+        if (action == 11) {
+            animation = &gShiekPlayingHarp3Anim;
+        } else if (action == 12) {
+            animation = &gShiekPlayingHarp2Anim;
+        } else if (action == 13) {
+            animation = &gShiekPlayingHarp4Anim;
+        } else {
+            animation = &gShiekPlayingHarp5Anim;
+            if (action == 23) {
+                animation = &gShiekPlayingHarpAnim;
+            }
+        }
+    }
+    return animation;
+}
+
+void EnXc_CalcXZAccel(EnXc* this) {
+    f32 timer = this->timer;
+    f32* speedXZ = &this->actor.speedXZ;
+
+    if (timer < 9.0f) {
+        *speedXZ = 0.0f;
+    } else {
+        if (timer < 3.0f) {
+            *speedXZ = (((kREG(2) * 0.01f) + 1.2f) / 3.0f) * (timer - 9.0f);
+        } else {
+            *speedXZ = (kREG(2) * 0.01f) + 1.2f;
+        }
+    }
+    Actor_MoveForward(&this->actor);
+}
+
+void func_80B3D644(EnXc* this) {
+    Actor_MoveForward(&this->actor);
+}
+
+void EnXc_CalcXZSpeed(EnXc* this) {
+    f32 timer = this->timer;
+    f32* speedXZ = &this->actor.speedXZ;
```

Comment:
Are either of these temps needed?

Reply:
Yeah they are. This one took forever to match